Add warning for invalid priority values in Group.set_priority()

emsilva opened the pull request:
## Summary
This PR addresses the TODO comment in `lib/ansible/inventory/group.py` by adding a proper warning when invalid priority values are passed to `Group.set_priority()`.

## Changes Made
- **Enhanced error handling**: Replace silent error handling with user-friendly warning messages
- **Comprehensive exception handling**: Catch both `TypeError` and `ValueError` exceptions for robust error handling
- **Preserve behavior**: Keep existing priority value when conversion fails, maintaining backward compatibility
- **Added test coverage**: Comprehensive test suite for both valid and invalid priority values
- **Added changelog fragment**: Documented the change for release notes

## Problem Solved
Previously, when invalid priority values were passed to `Group.set_priority()`, the method would silently fail without providing any feedback to users. This made it difficult to identify configuration issues.

## Solution
The method now displays clear warning messages that include:
- The invalid value that was provided
- The group name where the error occurred
- The specific error details from the conversion attempt

## Testing
- ✅ All existing tests pass
- ✅ Added comprehensive test coverage for both valid and invalid scenarios
- ✅ Manual testing confirms warnings are displayed correctly
- ✅ PEP8 compliance verified

## Examples
```python
# Valid priority (works as before)
group.set_priority(5)  # Sets priority to 5

# Invalid priority (now shows warning)
group.set_priority('invalid')  # Shows: [WARNING]: Invalid priority value 'invalid' for group 'test_group': invalid literal for int() with base 10: 'invalid'
```

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist
- [x] Code follows the project's style guidelines
- [x] Self-reviewed the changes
- [x] Added tests that prove the fix is effective
- [x] All tests pass locally
- [x] Added changelog fragment
- [x] Documentation updated (if applicable)

This improvement enhances the user experience by providing clear feedback when invalid priority values are used, helping users quickly identify and resolve configuration issues.

emsilva committed:
Add warning for invalid priority values in Group.set_priority()

This commit addresses the TODO comment in group.py by adding a proper warning
when invalid priority values are passed to Group.set_priority().

Changes:
- Replace silent error handling with user-friendly warning messages
- Catch both TypeError and ValueError exceptions for comprehensive error handling
- Preserve the existing priority value when conversion fails
- Add comprehensive test coverage for both valid and invalid priority values

The fix improves user experience by providing clear feedback when invalid
priority values are used, helping users identify configuration issues.

emsilva committed:
Add changelog fragment for group priority warning feature

Akasurde commented on lib/ansible/cli/config.py:
```diff
@@ -36,6 +36,9 @@
 _IGNORE_CHANGED = frozenset({'_terms', '_input'})
 
+# Settings that should be masked when displaying values
+_SENSITIVE_SETTINGS = frozenset({'token', 'password', 'client_secret'})
```
@emsilva, I think you mixed changes from https://github.com/ansible/ansible/pull/85467 in this PR.

Akasurde commented on test/units/inventory/test_group.py:
```diff
@@ -152,3 +152,40 @@ def test_ancestors_recursive_loop_safe(self):
+
+        # Test with negative numbers
+        group.set_priority(-1)
+        self.assertEqual(group.priority, -1)
```
```suggestion
        param_list = [
            (5, 5),  # int
            ('10', 10),  # string
            (-1, -1)  # negative number
        ]
        for priority, expected in param_list:
            group.set_priority(priority)
            self.assertEqual(group.priority, expected)
```

emsilva committed:
Improve test_set_priority_valid_values with parameterized approach

- Use param_list with tuples for more concise testing
- Addresses review feedback from Akasurde

Co-Authored-By: Akasurde <akasurde@redhat.com>

bcoca commented on lib/ansible/inventory/group.py:
```diff
@@ -266,6 +266,6 @@ def get_vars(self) -> dict[str, t.Any]:
-            # FIXME: warn about invalid priority
-            pass
+        except (TypeError, ValueError) as e:
+            display.warning(f"Invalid priority value '{priority}' for group '{self.name}': {e}")
```
use `error_as_warning` instead and just pass the exception in

bcoca replied:
Also the message should state that it falls back to the default

webknjaz commented on test/units/inventory/test_group.py:
```diff
@@ -152,3 +152,37 @@ def test_ancestors_recursive_loop_safe(self):
+        """Test that valid priority values are set correctly"""
+        group = Group('test_group')
+        
+        param_list = [
+            (5, 5),  # int
+            ('10', 10),  # string
+            (-1, -1)  # negative number
+        ]
```
Instead of having logic within tests, parametrize them. This can be either `@pytest.mark.parametrize` or `with self.subTest()`.

webknjaz commented on test/units/inventory/test_group.py:
```diff
@@ -152,3 +152,37 @@ def test_ancestors_recursive_loop_safe(self):
+        group = Group('test_group')
+        original_priority = group.priority
+
+        # Test with invalid string
+        group.set_priority('invalid')
+        self.assertEqual(group.priority, original_priority)  # Should remain unchanged
+
+        # Test with None
+        group.set_priority(None)
+        self.assertEqual(group.priority, original_priority)  # Should remain unchanged
+
+        # Test with dict
+        group.set_priority({'key': 'value'})
+        self.assertEqual(group.priority, original_priority)  # Should remain unchanged
+
+        # Test with list
+        group.set_priority(['item'])
+        self.assertEqual(group.priority, original_priority)  # Should remain unchanged
```
This is a bunch of individual tests, they should be in dedicated test methods. Also, the comments seem redundant.
